### PR TITLE
Add 'levels#next' to access next level

### PR DESCRIPTION
**Pull request by weiqingtoh**
After this, I can proceed to doing the views for the user badge. #632

### CODE REVIEW

File: app/models/course/level.rb [hunk 1]
```diff
@@ -20,6 +20,14 @@ def default_level?
     experience_points_threshold == 0
   end
 
+  # Returns the next higher level in the course
+  # nil is returned if current level is the highest level
+  #
+  # @return [Course::Level, nil]
```
Comment (lowjoel) [inline app/models/course/level.rb]: I think the syntax is `[Course::Level|nil]`

**Comment by lowjoel** [comment]
@weiqingtoh this looks fine currently, I'm just worried that the overall order of magnitude of calculating this might be prohibitively expensive at scale.

File: app/models/course/level.rb (hunk 1 above)
Comment (fonglh) [inline app/models/course/level.rb]: Looks like there should be multiple `@return` tags?
http://www.rubydoc.info/gems/yard/file/docs/Tags.md#return

Comment (lowjoel) [inline app/models/course/level.rb]: @fonglh good catch! Okay let's stick with multiple return tags then. My documentation is a mismash of yardoc, javadoc, doxygen...

**Pull request merged**